pkg/secrets/sealed_secrets.go

```diff
@@ -44,44 +46,72 @@ func replaceSecretsWithSealedSecrets(archivePath string, config map[string][]byt
 			return errors.Wrap(err, "failed to read file")
 		}
 
-		decoded, _, err := decode(contents, nil, nil)
-		if err != nil {
-			return nil
-		}
+		multiDocYaml := bytes.Split(contents, []byte("---\n"))
+		var secrets []byte
+		var nonSecrets []byte
+		for i := 0; i < len(multiDocYaml); i++ {
+			object := multiDocYaml[i]
+			if string(object) == "" {
+				continue
+			}
 
-		secret, ok := decoded.(*v1.Secret)
-		if !ok {
-			return nil
-		}
+			decoded, _, err := decode(object, nil, nil)
+			if err != nil {
+				nonSecrets = append(nonSecrets, []byte("---\n")...)
```

Comment:
i'd recommend following my suggestion above, since not all documents end with a new line. or you can use `\n---\n` instead